scripts/cleanup_analysis_entries.py

```diff
@@ -0,0 +1,259 @@
+    print(len(analyses), "total analyses")
+    print(len(ids_to_delete), "duplicates detected")
+
+    # example = duplicate_analyses[0]
+    # print(example)
+    # exit()
+
+    # throw warnings if the analyses['project'] doesn't match the bucket name, `cpg-{dataset}-{main/test}`
+    # check_all_analyses_belongs_to_project(analyses, project_id_map)
+
+    analysis_by_path = {a['output'] for a in analyses}
+    print(len(analysis_by_path), "unique paths found")
+ 
+    # TODO just write a function to check if paths are valid, too much dictionary passing in main
+    invalid_paths = []
+    for a in analysis_by_path:
+        a_json_str = a.replace("'","\"")
+        try:
+            a_dict = json.loads(a_json_str)
+            ap = str(a_dict['cram'])
+            print("dict read:", ap)
+        except json.decoder.JSONDecodeError:
+            ap = a
+            print("string read:", ap)
+
+        #if not ap.startswith('gs://'):
+        #    invalid_paths.append(ap)
+
+    # invalid_paths = [a['cram'] for a in analysis_by_path if not a.startswith('gs://')]
+    if invalid_paths != []:
+        # warnings.warn(f'{len(invalid_paths)} invalid paths found. Check log for details.')
+        print(f'{len(invalid_paths)} invalid paths found. Check log for details.')
+    # TODO: error handling for invalid paths
+
+    # let's check which files exist
+    path_existence = validate_paths(analysis_by_path)
+
+    print(f'{len(path_existence["valid"])} valid paths, {len(path_existence["invalid"])} invalid paths')
+
+    print('Analysis paths with no file discovered:', path_existence['invalid'])
+
+
+    # now do more stuff to check:
+    #   analysis-entries we have to delete because files don't exist
+    #   samples that DID have analysis and now don't
+    if dry_run:
+        print(f'Ready for ids to delete: {ids_to_delete}')
+    else:
+        # delete some analysis entries
+        pass
+
+def get_analyses_for_datasets(datasets, analysis_type) -> list[dict]:
+    # get all analysis-entries 
+    return aapi.query_analyses(
+            AnalysisQueryModel(
+                projects=datasets,
+                type=AnalysisType(analysis_type),
+                # status=AnalysisStatus('completed'),
+            )
+        )
+
+def find_invalid_analyses(analyses: list[dict]):
+    """
+    
+    
+    """
+    # flag analyses in a bad state
+    pass
+
+
+def get_path_components_from_path(path):
+    """
+    Return the {bucket_name}, {dataset}, and {subdir} for GS only paths
+
+    >>> get_bucket_name_path_from_path('gs://cpg-dataset-main/subfolder/subfolder2/my.cram')
+    {bucket_name:'cpg-dataset-main', dataset:'dataset', subdir:'subfolder/subfolder2'}
+    """
+    if path.startswith('gs://'):
+        short_path = path.removeprefix('gs://').split('/', maxsplit=1)
+        bucket_name = short_path[0]
+
+        dataset_with_bucket_type = bucket_name.removeprefix('cpg-')
+        bucket_type = dataset_with_bucket_type.split('-')[-1]
```

Comment:
we have single and double-suffixed buckets: [see](https://github.com/populationgenomics/team-docs/tree/main/storage_policies)

It would be good to slot in some test cases for `-upload`, `-analysis`, `-web` variants

Reply:
Alternatively if you process each dataset's analysis entries separately you can just keep hold of the dataset name, so you don't need to worry about how to identify

```
gs://cpg-acute-care-main-analysis/
vs.
gs://cpg-acute-care-main/
```